File: TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/inread/InReadRecyclerViewFragment.java
package tv.teads.teadssdkdemo.format.inread;

import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.greenrobot.eventbus.Subscribe;

import java.util.ArrayList;

import tv.teads.teadssdkdemo.R;
import tv.teads.teadssdkdemo.format.adapter.SimpleRecyclerViewAdapter;
import tv.teads.teadssdkdemo.utils.BaseFragment;
import tv.teads.teadssdkdemo.utils.ReloadEvent;

/**
 * InRead format within a RecyclerView
 * <p/>
 * Created by Hugo Gresse on 30/03/15.
 */
public class InReadRecyclerViewFragment extends BaseFragment {

    /**
     * The RecyclerView used in the application
     */
    private RecyclerView mRecyclerView;

    private SimpleRecyclerViewAdapter mAdapter;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_inread_recyclerview, container, false);

        // Retrieve recyclerView from layout
        mRecyclerView = rootView.findViewById(R.id.recyclerView);


        return rootView;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        // Set RecyclerView basic adapter
        setRecyclerViewAdapter(mRecyclerView);

    }

    private void setRecyclerViewAdapter(RecyclerView recyclerView) {
        ArrayList<String> data = new ArrayList<>();

        for (int i = 0; i < 50; i++) {
            data.add("Teads " + i);
        }

        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));

        mAdapter = new SimpleRecyclerViewAdapter(data, getPid(), 10);
        recyclerView.setAdapter(mAdapter);
    }

    @Subscribe
    @SuppressWarnings("unused")
    public void onReloadEvent(ReloadEvent event) {
        if (mAdapter != null) {
            mAdapter.reloadAd();
        }
    }

}

File: TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/inread/InReadRecyclerViewFragment.kt
package tv.teads.teadssdkdemo.format.inread

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import tv.teads.teadssdkdemo.databinding.FragmentRecyclerviewBinding
import tv.teads.teadssdkdemo.format.inread.adapter.SimpleRecyclerViewAdapter
import tv.teads.teadssdkdemo.utils.BaseFragment

/**
 * InRead format within a RecyclerView
 *
 */
class InReadRecyclerViewFragment : BaseFragment() {
    private lateinit var binding: FragmentRecyclerviewBinding
    private lateinit var adapter: SimpleRecyclerViewAdapter

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        binding = FragmentRecyclerviewBinding.inflate(layoutInflater)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        setRecyclerViewAdapter(binding.recyclerView)
    }

    private fun setRecyclerViewAdapter(recyclerView: RecyclerView) {
        recyclerView.layoutManager = LinearLayoutManager(activity)

        adapter = SimpleRecyclerViewAdapter(context, pid, getTitle())
        recyclerView.adapter = adapter
    }

    override fun getTitle(): String = "InRead Direct RecyclerView"
}
